From f9e7685844d81f36367d0ba222c6cf9b550f025b Mon Sep 17 00:00:00 2001
From: Pete Beardmore <pete.beardmore@msn.com>
Date: Tue, 10 Feb 2015 15:37:10 +0000
Subject: fix map option parsing for 'strictatime'

Added a trivial token length check to ensure the 'strictatime'
mount option is not consumed by string logic for the 'strict'
automount option.

Pre-patch, adding the 'strictatime' to a map effectively switched
strict mode (for multi-mount handling) on, and failed to pass the
'strictatime' option through to 'mount'.

This will have been an issue since 'strictatime' was introduced
for the 2.6.30 kernel.
---
 modules/parse_sun.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/modules/parse_sun.c b/modules/parse_sun.c
index b881ee9..86ee279 100644
--- a/modules/parse_sun.c
+++ b/modules/parse_sun.c
@@ -509,7 +509,8 @@ static int sun_mount(struct autofs_point *ap, const char *root,
 				fstype[typelen] = '\0';
 			} else if (_strncmp("nonstrict", cp, 9) == 0) {
 				nonstrict = 1;
-			} else if (_strncmp("strict", cp, 6) == 0) {
+			} else if (_strncmp("strict", cp, 6) == 0 &&
+				   comma - cp == 6) {
 				nonstrict = 0;
 			} else if (_strncmp("nobrowse", cp, 8) == 0 ||
 				   _strncmp("browse", cp, 6) == 0 ||
-- 
1.8.4

